### PR TITLE
feat(bobutils): add GFM table-cell escape helper

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

Two brain-repo idea-backlog generators each carry a copy of a GFM table-cell
escaper. An unescaped `|` splits a row into phantom columns, which makes
`idea_backlog_terminal_state_audit.py` refuse to rewrite the table and jams
`bob-retire-graduated-ideas.service` (2026-08-28, rows 1163/1164).

This adds `bobutils.markdown.escape_table_cell` (idempotent; skips already
backslash-escaped pipes) and `split_table_row_cells` (the consumer-side
counterpart). Producer and consumer share one regex so they cannot drift.
`str.split('|')` is the wrong oracle because it does not understand escaping;
the tests assert against the real splitter.

Brain-repo follow-up (after this merges and the submodule is bumped): point
`scripts/generate-backlog-ideas.py` and `scripts/seed-idea-backlog.py` at the
shared helper and drop the local copies.

## Test plan

- [x] `cd packages/bobutils && uv run pytest tests/test_markdown.py -q` — 8 passed
- [x] `make -C packages/bobutils typecheck` — clean

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
CI is green and the PR is mergeable. Self-merge is not eligible yet because the required Greptile review has not arrived; waiting for that review before re-running the merge gate.

File: packages/bobutils/src/bobutils/markdown.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+def split_table_row_cells(line: str) -> list[str]:
+    """Return stripped content cells for a single GFM table row.
+
+    Splits on unescaped pipes only. Leading/trailing delimiter pipes are
+    dropped so ``| a | b |`` → ``['a', 'b']``. This is the consumer-side
+    counterpart of :func:`escape_table_cell` — ``str.split('|')`` is the
+    wrong oracle because it does not understand escaping.
+    """
+    parts = _UNESCAPED_PIPE_RE.split(line)
+    content_cells = parts[1:-1] if line.endswith("|") else parts[1:]
```
Comment (TimeToBuildBob) [inline packages/bobutils/src/bobutils/markdown.py]: ❌ **P1** — The split logic drops the first cell when the row does not have a leading delimiter pipe. It always slices parts[1:] (or parts[1:-1]) without checking whether the line starts with '|'. For a row like "a | b |", parts = ["a ", " b ", ""], and since it ends with '|', it takes parts[1:-1] = [" b "], silently discarding the first cell "a". Similarly, for "a|b" (no leading or trailing pipe), parts = ["a","b"] and content_cells = parts[1:] = ["b"], losing "a". This yields an incorrect column count and can cause the consumer to write a malformed table, which is exactly the issue this helper is meant to prevent. The fix is to conditionally drop the first part only when line.startswith("|") and the last part only when line.endswith("|").

**Comment by TimeToBuildBob** [comment]
## 🤖 AI code review

Adds `bobutils.markdown` with `escape_table_cell` (escapes unescaped `|` in cell content, idempotent) and `split_table_row_cells` (splits a GFM table row on unescaped pipes, dropping delimiter pipes). Both share `_unescaped_pipe_indices` which handles backslash runs by parity. Adds a test suite covering escaping, splitting, idempotency, and backslash-run cases.

### Confidence Score: 5/5 — No findings on this head

✅ **No findings.** The diff looks correct to me on this pass.

<details><summary><b>Files changed (2)</b> — the diff as I read it</summary>

- `packages/bobutils/src/bobutils/markdown.py` — New module with two public functions and a private scanner for unescaped pipe indices.
- `packages/bobutils/tests/test_markdown.py` — New test suite for the escaping and splitting functions, including backslash-run edge cases.

</details>

<details><summary>Previous review passes</summary>

| commit | score | findings | engine | when |
| --- | --- | --- | --- | --- |
| `7fb94ab163d4` | 3/5 | 1 | llm | 2026-08-28 10:09 UTC |
| `591d762fedab` | 3/5 | 1 | llm | 2026-08-28 16:25 UTC |
| `2d0431d78db5` | 5/5 | 0 | llm | 2026-08-28 17:03 UTC |

</details>

<sub>Reviewed `2d0431d78db5` · `openrouter/deepseek/deepseek-v4-flash-0731` · `llm` engine · 189s · [about this reviewer](https://github.com/gptme/gptme-contrib/blob/master/docs/ai-reviewer.md)</sub>

<details><summary><sub>Maintainer commands</sub></summary>

<sub>`@TimeToBuildBob review` (own line) — fresh review · `@TimeToBuildBob fix` — a worker acts on the findings. Once per comment; 👀 = received.</sub>

</details>

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: packages/bobutils/src/bobutils/markdown.py (hunk 1 above)
Comment (TimeToBuildBob) [reply packages/bobutils/src/bobutils/markdown.py]: Fixed in 591d762f — `split_table_row_cells` now drops only empty leading/trailing split parts (the unescaped delimiter pipes). Rows without a leading `|`, and last cells that end in `\|`, keep their cells. Regression tests cover both cases.

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: packages/bobutils/src/bobutils/markdown.py (hunk 1 above)
Comment (TimeToBuildBob) [reply packages/bobutils/src/bobutils/markdown.py]: Follow-up in 2d0431d7 — the lookbehind missed even-backslash runs (`\\|`). Shared scanner now walks escape pairs so those pipes stay delimiters. The original parts[1:] slice is still gone.

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by TimeToBuildBob** [comment]
CI is green and MERGEABLE. Self-merge still blocked: the AI-review marker on `2d0431d` is 5/5 with no findings, but it is a 1-pass record and the gate requires ≥2 consensus passes. Greptile is dark (four triggers, no ack). A 2-of-3 consensus upgrade failed this session — `OPENROUTER_API_KEY_REVIEW` hit its daily limit (403). Retry after that key resets at the next UTC midnight. Did not merge or bypass the gate.

**Comment by TimeToBuildBob** [comment]
AI review is already current-head 5/5 on `2d0431d` (no findings). The `review_stale_vs_head` re-review dispatch is a false positive — the marker SHA matches HEAD.

Remaining self-merge gate: 1-pass consensus (`requested: 1`, need ≥2). Greptile is still dark (four triggers, no ack). `OPENROUTER_API_KEY_REVIEW` is daily-capped until 2026-08-29T00:00Z. Recheck already scheduled at 00:10Z for `--force` consensus upgrade then `self-merge-if-eligible`. Did not re-review, retrigger Greptile, or merge.
